src/prefect/utilities/collections.py

```diff
@@ -289,20 +283,18 @@ def visit_nested(expr):
         result = expr
 
     elif typ in (list, tuple, set):
-        items = await gather(*[visit_nested(o) for o in expr])
+        items = [visit_nested(o) for o in expr]
         result = typ(items) if return_data else None
 
     elif typ in (dict, OrderedDict):
         assert isinstance(expr, (dict, OrderedDict))  # typecheck assertion
         keys, values = zip(*expr.items()) if expr else ([], [])
-        keys = await gather(*[visit_nested(k) for k in keys])
-        values = await gather(*[visit_nested(v) for v in values])
+        keys = [visit_nested(k) for k in keys]
+        values = [visit_nested(v) for v in values]
         result = typ(zip(keys, values)) if return_data else None
```

Comment:
I wonder if these can be combined into one statement now now that the await/gather is removed?
```
typ((visit_nested(k), visit_nested(v)) for k, v in expr.items())
```

Reply:
I took a poke at this but we don't want to construct the new type unless `return_data` is `True`, we still need to visit all of the keys and values. I think I've got a simplification but it's still two lines